src/open-uri.c

```diff
@@ -600,10 +602,10 @@ handle_open_in_thread_func (GTask *task,
   if (latest_id != NULL)
     {
       /* Add extra options to the request for the backend */
-      g_variant_builder_add (&opts_builder,
-                             "{sv}",
-                             "last_choice",
-                             g_variant_new_string (latest_id));
+      g_variant_builder_add (&opts_builder, "{sv}", "last_choice", g_variant_new_string (latest_id));
+      g_variant_builder_add (&opts_builder, "{sv}", "content_type", g_variant_new_string (content_type));
+      if (basename)
+        g_variant_builder_add (&opts_builder, "{sv}", "filename", g_variant_new_string (basename));
     }
```

Comment:
Is there any reason why you would not want to send the content type and filename to the backend unless latest_id is set? It seems to me that sending them regardless would be useful to show the right application in the dialog (or to show the right mode of GS using `--search`)